### PR TITLE
NEW FEATURE: Adding Split, Survival, Clustering, FCMPLayer

### PR DESCRIPTION
**Pull request by wenyushi**
Please help me review the PR.
The PR contains the changes as follows:
1. Added more operators in layers.py, including Split, Survival, Clustering, FCMPLayer
2. Implemented all extractors for existing operators, including fcmp, segmentation, embeddingloss, cluster, survival, split. Besides, I propose to add a check for unsupported operators to be extracted from a table.
3. Testcases covering the changes.
4. Remove distance argument in embedding loss layer because it is a hidden option.

### CODE REVIEW

File: dlpy/layers.py [hunk 1]
```diff
@@ -2561,7 +2561,7 @@ class EmbeddingLoss(Layer):
     can_be_last_layer = True
     number_of_instances = 0
 
-    def __init__(self, name=None, distance="L2", margin=2, src_layers=None, **kwargs):
+    def __init__(self, name=None, margin=2, src_layers=None, **kwargs):
```
Comment (ethem-kinginthenorth) [inline dlpy/layers.py]: we remove this as we dont use it?

File: dlpy/layers.py [hunk 2]
```diff
@@ -2840,13 +2843,183 @@ def num_weights(self):
...
+        return None
+
+    @property
+    def num_weights(self):
+        return 0
+
+    @property
+    def output_size(self):
+        if self._output_size is None:
+            # 19w47 split layer can only take one source layer.
```
Comment (ethem-kinginthenorth) [inline dlpy/layers.py]: please use the official version number

File: dlpy/layers.py (hunk 1 above)
Comment (wenyushi) [reply dlpy/layers.py]: Yes, the hidden option should not be exposed. Yes, I ran the unit tests and will add a benchmark unit test for the layer.

File: dlpy/layers.py (hunk 2 above)
Comment (wenyushi) [reply dlpy/layers.py]: ok, will change.

**Pull request merged**